CMakeLists.txt

```diff
@@ -56,6 +56,7 @@ option(ENABLE_WERROR "Treat warnings as errors" OFF)
 option(ENABLE_NLS "Enable native language support" ON)
 option(INSTALL_DOC "Build/install documentation" ON)
 option(INSTALL_LIB "Install the library" OFF)
+option(DISABLE_DEPRECATED "Set 'disable deprecated' option for upstream packages, e.g. GTK" AUTO)
```

Comment:
Options are boolean by definition, `AUTO` isn't a valid value. Use `tr_auto_option()` instead.

Also minor, but maybe name `ENABLE_DEPRECATED` to be in line with other options.